qemu/tests/luks_image_over_qsd.py

```diff
@@ -0,0 +1,91 @@
+import socket
+
+from provider.qsd import QsdDaemonDev
+
+from virttest import error_context, env_process
+from virttest import qemu_storage
+from virttest import utils_numeric
+from virttest import data_dir
+from virttest.qemu_monitor import QMPCmdError
+
+
+# This decorator makes the test function aware of context strings
+@error_context.context_aware
+def run(test, params, env):
+    """
+        Export a luks image via QSD with NBD inet.
+        1. Create a luks image.
+        2. Start a QSD daemon, and export the luks image with NBD inet.
+        3. Check the info of the image over NBD.
+        4. Boot up a guest with the exported image as a data disk.
+        5. Check there is the data disk in the guest.
+        6. Connect to the QMP sock
+        6.1 Query the block exports
+        6.2 luks key management.
+        :param test: VT test object.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+    qsd = QsdDaemonDev("qsd1", params)
+    qsd.start_daemon()
+
+    luks_image_size = params['image_size_stg1']
+    luks_img_param = params.object_params('stg1')
+    luks_img = qemu_storage.QemuImg(luks_img_param, data_dir.get_data_dir(),
+                                    'stg1')
+    nbd_image_tag = params['nbd_image_tag']
+    nbd_image_params = params.object_params(nbd_image_tag)
+    localhost = socket.gethostname()
+    nbd_image_params['nbd_server'] = localhost if localhost else 'localhost'
+    qemu_img = qemu_storage.QemuImg(nbd_image_params, None, nbd_image_tag)
```

Comment:
Hit the error "ValueError: Empty image secret for nbd1" with the latest code, please help check it. Thanks.

Reply:
set image format to luks, it works. Thanks.

Reply:
hi @hellohellenmao , I am afraid this case should not depends on the imageformat=luks, because we specified the stg1 image format(luks) in cfg, even we use --imageformat=qcow2(image1 is qcow2), it should work well, so here we need to add image_secret_nbd1 = ${image_secret_stg1} in cfg (I thought I added this comment, but I didn't , sorry for that)

also cc @XueqiangWei